src/services/codefixes/importFixes.ts

```diff
@@ -153,16 +220,25 @@ namespace ts.codefix {
     }
 
     // Sorted with the preferred fix coming first.
-    const enum ImportFixKind { UseNamespace, ImportType, AddToExisting, AddNew }
-    type ImportFix = FixUseNamespaceImport | FixUseImportType | FixAddToExistingImport | FixAddNewImport;
+    const enum ImportFixKind { UseNamespace, JsdocTypeImport, AddToExisting, AddNew }
+    // These should not be combined as bitflags, but are given powers of 2 values to
+    // easily detect conflicts between `NotAllowed` and `Required` by giving them a unique sum.
+    // They're also ordered in terms of increasing priority for a fix-all scenario (see
+    // `reduceAddAsTypeOnlyValues`).
+    const enum AddAsTypeOnly {

```

Comment:
Consider

```suggestion
    const enum TypeOnlyModifierPresence {
```

or

```suggestion
    const enum TypeOnlyModifierAddition {
```

Reply:
In the case of a named import, the `type` modifier need not be present _on_ the named import; it could be present on the parent import declaration... afraid calling this “TypeOnlyModifierBlah” makes you focus on the wrong thing.